### ISSUE TITLE
Possible Regression in Fascination (1991)

### ISSUE THREAD

**Issue by Python-Exoproject** [dosbox-staging/dosbox-staging#3162]
### Are you using the latest Dosbox-Staging Version?

- [X] I have checked releases and am using the latest release.

### Different version than latest?

0.81.0-alpha

### What Operating System are you using?

Windows 11

### Have you checked that no other similar issue already exists?

- [X] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

Using exodos v6 release of the game staging 80.1 plays the CD version (using exodos, select option 1 to play using DOSBox, then option 2 for CD version. At the config screen press enter) no problem. However in 81.0 alpha you get the following screenshot after the config menu closes:
![image](https://github.com/dosbox-staging/dosbox-staging/assets/67406191/80e02df4-db81-4e59-9da5-44b7b9400315)

### Steps to reproduce the behaviour.

As above

### Download URL of affected game or software

exodos v6

### Your configuration

```ini
exodos v6
```

### Provide a Log

```text
Nothing relevant
```

### Code of Conduct & Contributing Guidelines

- [X] Yes, I agree.

**Comment by weirddan455** [comment]
Confirmed regression not using exodos.  Starting a bisect to see if I can narrow this down.

**Comment by weirddan455** [comment]
```
c4a6c7db9abe3ebf84953744794b4b82266732e9 is the first bad commit
commit c4a6c7db9abe3ebf84953744794b4b82266732e9
Author: kcgen <kcgen@users.noreply.github.com>
Date:   Sat Mar 4 15:16:40 2023 -0800

    Make use of the new DOS device append routine in MSCDEX

 src/dos/dos_mscdex.cpp | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)
```

**Comment by johnnovak** [comment]
Paging

> ```
> c4a6c7db9abe3ebf84953744794b4b82266732e9 is the first bad commit
> commit c4a6c7db9abe3ebf84953744794b4b82266732e9
> Author: kcgen <kcgen@users.noreply.github.com>
> Date:   Sat Mar 4 15:16:40 2023 -0800
> 
>     Make use of the new DOS device append routine in MSCDEX
> 
>  src/dos/dos_mscdex.cpp | 10 +---------
>  1 file changed, 1 insertion(+), 9 deletions(-)
> ```

Paging @kcgen 😎

**Comment by weirddan455** [comment]
Also confirmed `git revert c4a6c7db9abe3ebf84953744794b4b82266732e9` on main fixes the bug.  It looks like this was meant to be an abstraction and there's probably some subtle bug in there (it's dealing with some DOS linked list).  I'll look into this more later unless @kcgen gets there first.

**Comment by kcgen** [comment]
Thanks for helping @weirddan455, and yes please - if you can keep running with it, I'm am still finding time to chip away on the powerpc and trying to keep my head in that space.

Looks like that commit was part of PR #2310, which itself was to fix a mount-order lockup bug (#2304). So if it's possible to repro this lockup bug and still fix this issue w/ Fascination, then it's win-win :-)
